references to vertx-template replaced with validation-service

katirp opened the pull request.

ksclarke reviewed (commented):
There are some additional references to template (I realized I was too specific in the ticket) that should be renamed too:

pom.xml:    <vertx.verticle>info.freelibrary.vertx.template.verticles.MainVerticle</vertx.verticle>
pom.xml:              <mediaTypePackage>info.freelibrary.vertx.template</mediaTypePackage>
pom.xml:                        <include>template.yaml</include>
src/test/java/info/freelibrary/vertx/template/verticles/MainVerticleTest.java:package info.freelibrary.vertx.template.verticles;
src/test/java/info/freelibrary/vertx/template/verticles/MainVerticleTest.java:import info.freelibrary.vertx.template.Config;
src/test/java/info/freelibrary/vertx/template/verticles/MainVerticleTest.java:import info.freelibrary.vertx.template.Op;
src/main/resources/template_messages.xml:  <entry key="message-class-name">info.freelibrary.vertx.template.MessageCodes</entry>
src/main/resources/template.yaml:  - url: http://github.com/ksclarke/template
src/main/java/info/freelibrary/vertx/template/Op.java:package info.freelibrary.vertx.template;
src/main/java/info/freelibrary/vertx/template/JsonKeys.java:package info.freelibrary.vertx.template;
src/main/java/info/freelibrary/vertx/template/handlers/StatusHandler.java:package info.freelibrary.vertx.template.handlers;
src/main/java/info/freelibrary/vertx/template/handlers/StatusHandler.java:import static info.freelibrary.vertx.template.MediaType.APPLICATION_JSON;
src/main/java/info/freelibrary/vertx/template/handlers/StatusHandler.java:import info.freelibrary.vertx.template.JsonKeys;
src/main/java/info/freelibrary/vertx/template/verticles/MainVerticle.java:package info.freelibrary.vertx.template.verticles;
src/main/java/info/freelibrary/vertx/template/verticles/MainVerticle.java:import info.freelibrary.vertx.template.Config;
src/main/java/info/freelibrary/vertx/template/verticles/MainVerticle.java:import info.freelibrary.vertx.template.MessageCodes;
src/main/java/info/freelibrary/vertx/template/verticles/MainVerticle.java:import info.freelibrary.vertx.template.Op;
src/main/java/info/freelibrary/vertx/template/verticles/MainVerticle.java:import info.freelibrary.vertx.template.handlers.StatusHandler;
src/main/java/info/freelibrary/vertx/template/verticles/MainVerticle.java:    private static final String API_SPEC = "src/main/resources/template.yaml";
src/main/java/info/freelibrary/vertx/template/Config.java:package info.freelibrary.vertx.template;

The project package (at `src/main/java`) can be named `edu.ucla.library.validation`. The standalone `template` (like in `template.yaml`) can be named `validation`. In the README, it can just be `Validation Service` for now (with more details to be added later).

ksclarke commented:
We also like to type a little in the PR comment to tell the reviewers what was changed in the PR... for instance:

* Updated references to vertx-template to validation-service
* Updated package name
* Updated OpenAPI spec file name

etc.

This PR is pretty straightforward but, if there are explanations or reasons for changes that will make the reviewer's job easier, we also like to include those. It's basically just a roadmap for the reviewer to know what they're looking at...

katirp commented:
New commit with changes: 

- Updated references to vertx-template to validation-service
- Updated references to template to validation, ex. OpenAPI spec file name
- Updated references to ksclarke to uclalibrary
- Updated package name to 'edu.ucla.library.validation'
- Updated pom.xml file with new naming

ksclarke commented on pom.xml:
```diff
@@ -193,7 +193,7 @@
             </goals>
             <configuration>
-              <mediaTypePackage>info.freelibrary.vertx.template</mediaTypePackage>
+              <mediaTypePackage>info.freelibrary.edu.ucla.library.validation</mediaTypePackage>
```
I think this should just be `edu.ucla.library.validation` instead of having the freelibrary stuff prepended to it.

ksclarke commented on src/main/java/edu/ucla/library/validation/handlers/StatusHandler.java:
```diff
@@ -1,11 +1,11 @@
+package edu.ucla.library.validation.handlers;
 
-import static info.freelibrary.vertx.template.MediaType.APPLICATION_JSON;
+import static info.freelibrary.edu.ucla.library.validation.MediaType.APPLICATION_JSON;
```
I added another comment about this, but you'll need to update this when you remove the `info.freelibrary` from the config in the POM too.

ksclarke reviewed (commented):
Would you also add `.DS_Store` to the project's .gitignore since that's a Mac specific thing that's not relevant to the project?

katirp commented:
- updated .gitignore to include mac specific files
- removed unnecessary info.freelibrary prefixes

katirp commented:
New commit to remove old .DS-store files: 
src/.DS_Store
src/main/.DS_Store
src/main/java/.DS_Store
src/main/java/edu/.DS_Store
src/main/java/edu/ucla/.DS_Store
src/main/java/edu/ucla/library/.DS_Store
